### ISSUE TITLE
SCIP solution not returned when total node limit reached

### ISSUE THREAD

**Issue by arvoelke** [cvxpy/cvxpy#2278]
**Describe the bug**
A solution is not returned when the SCIP status is "solving was interrupted [total node limit reached]", even when it has found a number of suboptimal solutions. This seems like a bug because it *will* return a solution if the solver is interrupted for other reasons, such as hitting a timeout.

**To Reproduce**
Run the below code twice, once with `be_good = False` and again with `be_good = True`:
1. When `be_good = False`, it runs for ~15 seconds and then hits the total node limit and raises an exception. No solution is returned despite it finding 17 suboptimal solutions.
2. When `be_good = True`, it runs for ~10 seconds and hits the timeout. A solution is available despite spending less time and finding the ~same solutions.

```python
import cvxpy as cp
import numpy as np

be_good = False  # set this to True to work-around the bug

if be_good:
    # Solution is accessible in this case.
    scip_params = {"limits/totalnodes": 5, "limits/time": 10}
else:
    # Solution is *not* accessible even though it spends more time and finds the same solutions.
    scip_params = {"limits/totalnodes": 5}

m = 100
n = 5
h = 1000

rng = np.random.RandomState(seed=0)
A = rng.randint(1, 10, size=(h, m))

X = cp.Variable((m, n), boolean=True)
c = cp.Variable(1)

objective = cp.Minimize(c)
constraints = [
    cp.sum(X, axis=1) == 1,
    A @ X <= c
]

prob = cp.Problem(objective, constraints)
prob.solve(solver=cp.SCIP, verbose=True, scip_params=scip_params)
print(prob.status, c.value)
```

**Expected behavior**
When the SCIP status is "solving was interrupted [total node limit reached]" I expect the best solution to be unpacked and returned, like it is in the case of "solving was interrupted [time limit reached]".

**Output**
```
===============================================================================
                                     CVXPY                                     
                                     v1.4.1                                    
===============================================================================
(CVXPY) Oct 30 02:01:37 PM: Your problem has 501 variables, 2 constraints, and 0 parameters.
(CVXPY) Oct 30 02:01:37 PM: It is compliant with the following grammars: DCP, DQCP
(CVXPY) Oct 30 02:01:37 PM: (If you need to solve this problem multiple times, but with different data, consider using parameters.)
(CVXPY) Oct 30 02:01:37 PM: CVXPY will first compile your problem; then, it will invoke a numerical solver to obtain a solution.
(CVXPY) Oct 30 02:01:37 PM: Your problem is compiled with the CPP canonicalization backend.
-------------------------------------------------------------------------------
                                  Compilation                                  
-------------------------------------------------------------------------------
(CVXPY) Oct 30 02:01:37 PM: Compiling problem (target solver=SCIP).
(CVXPY) Oct 30 02:01:37 PM: Reduction chain: Dcp2Cone -> CvxAttr2Constr -> ConeMatrixStuffing -> SCIP
(CVXPY) Oct 30 02:01:37 PM: Applying reduction Dcp2Cone
(CVXPY) Oct 30 02:01:37 PM: Applying reduction CvxAttr2Constr
(CVXPY) Oct 30 02:01:37 PM: Applying reduction ConeMatrixStuffing
(CVXPY) Oct 30 02:01:37 PM: Applying reduction SCIP
(CVXPY) Oct 30 02:01:37 PM: Finished problem compilation (took 1.202e-01 seconds).
-------------------------------------------------------------------------------
                                Numerical solver                               
-------------------------------------------------------------------------------
(CVXPY) Oct 30 02:01:37 PM: Invoking solver SCIP  to obtain a solution.
presolving:
   (0.2s) running MILP presolver
   (0.3s) MILP presolver (2 rounds): 0 aggregations, 0 fixings, 1 bound changes
(round 1, exhaustive) 0 del vars, 0 del conss, 0 add conss, 2 chg bounds, 0 chg sides, 0 chg coeffs, 100 upgd conss, 0 impls, 100 clqs
   (1.6s) probing cycle finished: starting next cycle
   (1.7s) symmetry computation started: requiring (bin +, int -, cont +), (fixed: bin -, int +, cont -)
   (1.9s) symmetry computation finished: 4 generators found (max: 1500, log10 of symmetry group size: 2.1)
(round 2, exhaustive) 11 del vars, 0 del conss, 1 add conss, 2 chg bounds, 0 chg sides, 0 chg coeffs, 100 upgd conss, 500 impls, 99 clqs
(round 3, fast)       13 del vars, 2 del conss, 1 add conss, 5 chg bounds, 0 chg sides, 0 chg coeffs, 100 upgd conss, 501 impls, 98 clqs
   (2.6s) probing cycle finished: starting next cycle
presolving (4 rounds: 4 fast, 3 medium, 3 exhaustive):
 13 deleted vars, 2 deleted constraints, 1 added constraints, 5 tightened bounds, 0 added holes, 0 changed sides, 0 changed coefficients
 600 implications, 111 cliques
presolved problem has 489 variables (488 bin, 0 int, 1 impl, 0 cont) and 5099 constraints
     98 constraints of type <setppc>
   5000 constraints of type <linear>
      1 constraints of type <orbitope>
transformed objective value is always integral (scale: 1)
Presolving Time: 2.62

 time | node  | left  |LP iter|LP it/n|mem/heur|mdpt |vars |cons |rows |cuts |sepa|confs|strbr|  dualbound   | primalbound  |  gap   | compl. 
p 2.8s|     1 |     0 |     0 |     - |  clique|   0 | 489 |5099 |5098 |   0 |  0 |   0 |   0 | 9.000000e+00 | 5.840000e+02 |6388.89%| unknown
p 2.9s|     1 |     0 |     0 |     - |   locks|   0 | 489 |5099 |5098 |   0 |  0 |   0 |   0 | 9.000000e+00 | 5.610000e+02 |6133.33%| unknown
p 3.1s|     1 |     0 |     2 |     - | vbounds|   0 | 489 |5099 |5098 |   0 |  0 |   2 |   0 | 9.000000e+00 | 5.580000e+02 |6100.00%| unknown
p 3.2s|     1 |     0 |     2 |     - | vbounds|   0 | 489 |5099 |5098 |   0 |  0 |   2 |   0 | 9.000000e+00 | 5.540000e+02 |6055.56%| unknown
  5.2s|     1 |     0 |  4495 |     - |   303M |   0 | 489 |5099 |5098 |   0 |  0 |   2 |   0 | 1.168000e+02 | 5.540000e+02 | 374.32%| unknown
r 5.3s|     1 |     0 |  4512 |     - |shifting|   0 | 489 |5099 |5098 |   0 |  0 |   2 |   0 | 1.170000e+02 | 2.130000e+02 |  82.05%| unknown
  5.3s|     1 |     0 |  4512 |     - |   303M |   0 | 489 |5099 |5098 |   0 |  0 |   2 |   0 | 1.170000e+02 | 2.130000e+02 |  82.05%| unknown
  5.3s|     1 |     0 |  4512 |     - |   303M |   0 | 489 |5099 |5098 |   0 |  0 |   2 |   0 | 1.170000e+02 | 2.130000e+02 |  82.05%| unknown
  5.8s|     1 |     0 |  4520 |     - |   305M |   0 | 489 |5099 |5100 |   2 |  1 |   2 |   0 | 1.170000e+02 | 2.130000e+02 |  82.05%| unknown
r 5.9s|     1 |     0 |  4520 |     - |shifting|   0 | 489 |5099 |5100 |   2 |  1 |   2 |   0 | 1.170000e+02 | 1.980000e+02 |  69.23%| unknown
  5.9s|     1 |     0 |  4563 |     - |   309M |   0 | 489 |5099 |5101 |   3 |  2 |   2 |   0 | 1.170000e+02 | 1.980000e+02 |  69.23%| unknown
r 6.0s|     1 |     0 |  4563 |     - |shifting|   0 | 489 |5099 |5101 |   3 |  2 |   2 |   0 | 1.170000e+02 | 1.970000e+02 |  68.38%| unknown
  6.0s|     1 |     0 |  4563 |     - |   309M |   0 | 489 |5099 |5101 |   3 |  2 |   2 |   0 | 1.170000e+02 | 1.970000e+02 |  68.38%| unknown
  6.0s|     1 |     0 |  4563 |     - |   309M |   0 | 489 |5099 |5101 |   3 |  2 |   2 |   0 | 1.170000e+02 | 1.970000e+02 |  68.38%| unknown
  6.1s|     1 |     0 |  4598 |     - |   312M |   0 | 489 |5099 |5102 |   4 |  3 |   2 |   0 | 1.170000e+02 | 1.970000e+02 |  68.38%| unknown
 time | node  | left  |LP iter|LP it/n|mem/heur|mdpt |vars |cons |rows |cuts |sepa|confs|strbr|  dualbound   | primalbound  |  gap   | compl. 
r 6.2s|     1 |     0 |  4598 |     - |shifting|   0 | 489 |5099 |5102 |   4 |  3 |   2 |   0 | 1.170000e+02 | 1.960000e+02 |  67.52%| unknown
  6.3s|     1 |     0 |  4648 |     - |   315M |   0 | 489 |5099 |5103 |   5 |  4 |   2 |   0 | 1.170000e+02 | 1.960000e+02 |  67.52%| unknown
r 6.4s|     1 |     0 |  4648 |     - |shifting|   0 | 489 |5099 |5103 |   5 |  4 |   2 |   0 | 1.170000e+02 | 1.950000e+02 |  66.67%| unknown
  6.4s|     1 |     0 |  4648 |     - |   315M |   0 | 489 |5099 |5103 |   5 |  4 |   2 |   0 | 1.170000e+02 | 1.950000e+02 |  66.67%| unknown
  6.4s|     1 |     0 |  4648 |     - |   315M |   0 | 489 |5099 |5103 |   5 |  4 |   2 |   0 | 1.170000e+02 | 1.950000e+02 |  66.67%| unknown
  6.5s|     1 |     0 |  4690 |     - |   319M |   0 | 489 |5099 |5104 |   6 |  5 |   2 |   0 | 1.170000e+02 | 1.950000e+02 |  66.67%| unknown
r 6.6s|     1 |     0 |  4690 |     - |shifting|   0 | 489 |5099 |5104 |   6 |  5 |   2 |   0 | 1.170000e+02 | 1.940000e+02 |  65.81%| unknown
  6.7s|     1 |     0 |  4743 |     - |   323M |   0 | 489 |5099 |5105 |   7 |  6 |   2 |   0 | 1.170000e+02 | 1.940000e+02 |  65.81%| unknown
r 6.8s|     1 |     0 |  4743 |     - |shifting|   0 | 489 |5099 |5105 |   7 |  6 |   2 |   0 | 1.170000e+02 | 1.930000e+02 |  64.96%| unknown
  6.8s|     1 |     0 |  4743 |     - |   323M |   0 | 489 |5099 |5105 |   7 |  6 |   2 |   0 | 1.170000e+02 | 1.930000e+02 |  64.96%| unknown
  6.8s|     1 |     0 |  4743 |     - |   323M |   0 | 489 |5099 |5105 |   7 |  6 |   2 |   0 | 1.170000e+02 | 1.930000e+02 |  64.96%| unknown
  6.9s|     1 |     0 |  4803 |     - |   325M |   0 | 489 |5099 |5107 |   9 |  7 |   2 |   0 | 1.170000e+02 | 1.930000e+02 |  64.96%| unknown
r 7.0s|     1 |     0 |  4803 |     - |shifting|   0 | 489 |5099 |5107 |   9 |  7 |   2 |   0 | 1.170000e+02 | 1.920000e+02 |  64.10%| unknown
  7.1s|     1 |     0 |  4862 |     - |   329M |   0 | 489 |5099 |5108 |  10 |  8 |   2 |   0 | 1.170000e+02 | 1.920000e+02 |  64.10%| unknown
r 7.1s|     1 |     0 |  4862 |     - |shifting|   0 | 489 |5099 |5108 |  10 |  8 |   2 |   0 | 1.170000e+02 | 1.910000e+02 |  63.25%| unknown
 time | node  | left  |LP iter|LP it/n|mem/heur|mdpt |vars |cons |rows |cuts |sepa|confs|strbr|  dualbound   | primalbound  |  gap   | compl. 
  7.2s|     1 |     0 |  4862 |     - |   329M |   0 | 489 |5099 |5108 |  10 |  8 |   2 |   0 | 1.170000e+02 | 1.910000e+02 |  63.25%| unknown
  7.2s|     1 |     0 |  4862 |     - |   329M |   0 | 489 |5099 |5108 |  10 |  8 |   2 |   0 | 1.170000e+02 | 1.910000e+02 |  63.25%| unknown
  7.3s|     1 |     0 |  4901 |     - |   333M |   0 | 489 |5099 |5109 |  11 |  9 |   2 |   0 | 1.170000e+02 | 1.910000e+02 |  63.25%| unknown
r 7.4s|     1 |     0 |  4901 |     - |shifting|   0 | 489 |5099 |5109 |  11 |  9 |   2 |   0 | 1.170000e+02 | 1.900000e+02 |  62.39%| unknown
  7.5s|     1 |     0 |  4975 |     - |   335M |   0 | 489 |5099 |5110 |  12 | 10 |   2 |   0 | 1.170000e+02 | 1.900000e+02 |  62.39%| unknown
r 7.6s|     1 |     0 |  4975 |     - |shifting|   0 | 489 |5099 |5110 |  12 | 10 |   2 |   0 | 1.170000e+02 | 1.890000e+02 |  61.54%| unknown
  7.6s|     1 |     0 |  4975 |     - |   335M |   0 | 489 |5099 |5110 |  12 | 10 |   2 |   0 | 1.170000e+02 | 1.890000e+02 |  61.54%| unknown
  7.6s|     1 |     0 |  4975 |     - |   335M |   0 | 489 |5099 |5110 |  12 | 10 |   2 |   0 | 1.170000e+02 | 1.890000e+02 |  61.54%| unknown
  7.6s|     1 |     0 |  4980 |     - |   336M |   0 | 489 |5099 |5111 |  13 | 11 |   2 |   0 | 1.170000e+02 | 1.890000e+02 |  61.54%| unknown
r 7.7s|     1 |     0 |  4980 |     - |shifting|   0 | 489 |5099 |5111 |  13 | 11 |   2 |   0 | 1.170000e+02 | 1.880000e+02 |  60.68%| unknown
  7.7s|     1 |     0 |  4980 |     - |   336M |   0 | 489 |5099 |5111 |  13 | 11 |   2 |   0 | 1.170000e+02 | 1.880000e+02 |  60.68%| unknown
d 9.3s|     1 |     0 |  9417 |     - |conflict|   0 | 489 |5099 |5111 |   0 | 13 |   2 |   0 | 1.170000e+02 | 1.360000e+02 |  16.24%| unknown
  9.5s|     1 |     0 |  9429 |     - |   343M |   0 | 489 |5099 |5111 |  13 | 13 |   2 |   0 | 1.170000e+02 | 1.360000e+02 |  16.24%| unknown
  9.6s|     1 |     0 |  9432 |     - |   343M |   0 | 489 |5099 |5112 |  14 | 15 |   2 |   0 | 1.170000e+02 | 1.360000e+02 |  16.24%| unknown
 13.2s|     1 |     2 |  9432 |     - |   343M |   0 | 489 |5099 |5112 |  14 | 15 |   2 |  22 | 1.170000e+02 | 1.360000e+02 |  16.24%| unknown

SCIP Status        : solving was interrupted [total node limit reached]
Solving Time (sec) : 13.65
Solving Nodes      : 5
Primal Bound       : +1.36000000000000e+02 (17 solutions)
Dual Bound         : +1.17000000000000e+02
Gap                : 16.24 %
---------------------------------------------------------------------------
SolverError                               Traceback (most recent call last)
Cell In[21], line 32
     26 constraints = [
     27     cp.sum(X, axis=1) == 1,
     28     A @ X <= c
     29 ]
     31 prob = cp.Problem(objective, constraints)
---> 32 prob.solve(solver=cp.SCIP, verbose=True, scip_params=scip_params)
     33 print(prob.status, c.value)

File ~/opt/anaconda3/envs/*/lib/python3.10/site-packages/cvxpy/problems/problem.py:503, in Problem.solve(self, *args, **kwargs)
    501 else:
    502     solve_func = Problem._solve
--> 503 return solve_func(self, *args, **kwargs)

File ~/opt/anaconda3/envs/*/lib/python3.10/site-packages/cvxpy/problems/problem.py:1086, in Problem._solve(self, solver, warm_start, verbose, gp, qcp, requires_grad, enforce_dpp, ignore_dpp, canon_backend, **kwargs)
   1084 end = time.time()
   1085 self._solve_time = end - start
-> 1086 self.unpack_results(solution, solving_chain, inverse_data)
   1087 if verbose:
   1088     print(_FOOTER)

File ~/opt/anaconda3/envs/*/lib/python3.10/site-packages/cvxpy/problems/problem.py:1411, in Problem.unpack_results(self, solution, chain, inverse_data)
   1409     warnings.warn(INF_OR_UNB_MESSAGE)
   1410 if solution.status in s.ERROR:
-> 1411     raise error.SolverError(
   1412             "Solver '%s' failed. " % chain.solver.name() +
   1413             "Try another solver, or solve with verbose=True for more "
   1414             "information.")
   1416 self.unpack(solution)
   1417 self._solver_stats = SolverStats.from_dict(self._solution.attr,
   1418                                  chain.solver.name())

SolverError: Solver 'SCIP' failed. Try another solver, or solve with verbose=True for more information.
```

**Version**
 - OS: macOS Monterey 12.5.1
 - CVXPY Version: 1.4.1

**Additional context**
Adding a timeout is not a suitable workaround for our use case as it leads to nondeterminism.

**Comment by SteveDiamond** [comment]
Good point! Do you want to try doing a pull request with a fix? It should be a small change to scip_conif.py

**Comment by arvoelke** [comment]
Thanks for the lead. I tried that out in https://github.com/cvxpy/cvxpy/pull/2279 and it does indeed fix our problem. Please let me know in that PR if and how to go about unit testing if required.

**Comment by SteveDiamond** [comment]
Great! We ran into lots of issues trying to unit test the time limit for SCIP, so no need for tests.

**Comment by arvoelke** [comment]
Fixed by #2279 which has been merged in. Thanks again.
